adapters/handlers/rest/authz/handlers_authz.go

```diff
@@ -85,6 +85,10 @@ func (h *authZHandlers) createRole(params authz.CreateRoleParams, principal *mod
 		return authz.NewCreateRoleBadRequest().WithPayload(cerrors.ErrPayloadFromSingleErr(errors.New("role name is invalid")))
 	}
 
+	if err := isRootRole(*params.Body.Name); err != nil {
+		return authz.NewCreateRoleBadRequest().WithPayload(cerrors.ErrPayloadFromSingleErr(errors.New("role name is invalid")))
+	}
```

Comment:
I think this is already being checked several lines below ('root' being one of the `BuiltinRoles`)

https://github.com/weaviate/weaviate/blob/b86330e680ccc584057d1ab6a40282632cf88f45/adapters/handlers/rest/authz/handlers_authz.go#L78-L80